generators/common/templates/devcontainer/Dockerfile.ejs

```
@@ -0,0 +1,25 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.209.6/containers/java/.devcontainer/base.Dockerfile
+
+# [Choice] Java version (use -bullseye variants on local arm64/Apple Silicon): 11, 17, 11-bullseye, 17-bullseye, 11-buster, 17-buster
+ARG VARIANT="11"
+FROM mcr.microsoft.com/vscode/devcontainers/java:0-${VARIANT}
+
+# [Option] Install Maven
+ARG INSTALL_MAVEN="false"
+ARG MAVEN_VERSION=""
+# [Option] Install Gradle
+ARG INSTALL_GRADLE="false"
+ARG GRADLE_VERSION=""
+RUN if [ "${INSTALL_MAVEN}" = "true" ]; then su vscode -c "umask 0002 && . /usr/local/sdkman/bin/sdkman-init.sh && sdk install maven \"${MAVEN_VERSION}\""; fi \
+    && if [ "${INSTALL_GRADLE}" = "true" ]; then su vscode -c "umask 0002 && . /usr/local/sdkman/bin/sdkman-init.sh && sdk install gradle \"${GRADLE_VERSION}\""; fi
+
+# [Choice] Node.js version: none, lts/*, 16, 14, 12, 10
+ARG NODE_VERSION="none"
+RUN if [ "${NODE_VERSION}" != "none" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
+
+# [Optional] Uncomment this section to install additional OS packages.
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
+
+# [Optional] Uncomment this line to install global node packages.
+# RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && npm install -g <your-package-here>" 2>&1
```

IMO we can keep the file clean, usually, our generated project doesn't need more tools.
We provide wrapper for maven, Gradle and npm.
So, I am wondering if this works:
```suggestion
RUN su ln -s /usr/local/bin/npm ${containerWorkspaceFolder}/npmw
  && su ln -s /usr/local/bin/mvn ${containerWorkspaceFolder}/mvnw
  && su ln -s /usr/local/bin/gradle ${containerWorkspaceFolder}/gradlew
```

---

The Dockerfile is the virtualisation for the IDE running the devcontainer. The paths are defined by CodeSpaces/Remote Containers [here](https://docs.github.com/en/codespaces/getting-started/deep-dive#running-your-application). This file was autogenerated using the Codespaces extension. It works not only in GitHub Codespaces, but also in VS Code dev containers, GitPod and others.